test/core/common/buffer_test.cpp

```diff
@@ -0,0 +1,51 @@
+#include <gtest/gtest.h>
+#include "common/buffer.hpp"
+
+using namespace kagome::common;
+using namespace std::string_literals;
+
+TEST(Common, Buffer_Put) {
+  Buffer b;
+  ASSERT_EQ(b.size(), 0);
+
+  auto hex = b.to_hex();
+  ASSERT_EQ(hex, ""s);
+
+  b.put("hello");
+  ASSERT_EQ(b.size(), 5);
+
+  b.put_uint8(1);
+  ASSERT_EQ(b.size(), 6);
+
+  b.put_uint32(0);
+  ASSERT_EQ(b.size(), 10);
+
+  b.put_uint64(0);
+  ASSERT_EQ(b.size(), 18);
+
+  b.put(std::vector<uint8_t>{1, 2, 3, 4, 5});
+  ASSERT_EQ(b.size(), 23);
+
```

Comment:
Please, check also a hex representation after you manually added those single bytes